### PR TITLE
ENH: Improve parquet io

### PR DESCRIPTION
**Pull request by bashtage**
Synchronize with upstream
Eliminate redundant code

- [X] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

File: pandas-stubs/core/frame.pyi [hunk 1]
```diff
@@ -275,23 +275,23 @@ class DataFrame(NDFrame, OpsMixin):
...
-        compression: Literal["snappy", "gzip", "brotli"] = ...,
-        index: _bool | None = ...,
-        partition_cols: list | None = ...,
+        path: FilePath | WriteBuffer[bytes],
+        engine: ParquetEngine = ...,
+        compression: Literal["snappy", "gzip", "brotli"] | None = ...,
+        index: bool | None = ...,
+        partition_cols: list[HashableT] | None = ...,
+        storage_options: StorageOptions = ...,
         **kwargs,
```
Comment (Dr-Irv) [inline pandas-stubs/core/frame.pyi:284]: I see that you typed `**kwargs` as `Any` in `read_parquet()`, so why not do it here (or not type it there)?

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
One small question about consistency in typing `**kwargs`, otherwise looks OK

File: pandas-stubs/core/frame.pyi (hunk 1 above)
Comment (bashtage) [reply pandas-stubs/core/frame.pyi]: Added, thanks.

**Comment by bashtage** [comment]
Should be ready once green.

**Review by Dr-Irv — APPROVED** [review]
thanks @bashtage

**Pull request merged**